common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/dialog/TextAlertDialogBuilder.kt

```diff
@@ -111,11 +111,9 @@ class TextAlertDialogBuilder(val context: Context) : DaxAlertDialog {
             .setCancelable(isCancellable)
             .apply {
                 if (isCancellable) {
-                    setOnKeyListener(
-                        BackKeyListener {
-                            listener.onDialogCancelled()
-                        },
-                    )
+                    setOnCancelListener {
+                        listener.onDialogCancelled()
+                    }
```

Comment:
Wondering if what David wanted here was setting the back key in case it's not cancellable so the experience is not that intrusive 🤔 
```suggestion
             if (!isCancellable) {       
                    setOnKeyListener(
                        BackKeyListener {
                            listener.onDialogCancelled()
                        },
                     )
```

Reply:
I've created https://app.asana.com/0/1202857801505092/1204448904099019/f to capture the question. We can review it, but not blocking the fix.